ScoutHelper/Managers/SirenManager.cs

```diff
@@ -0,0 +1,180 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.DataAnnotations;
+using System.IO;
+using System.Linq;
+using System.Numerics;
+using System.Text;
+using CSharpFunctionalExtensions;
+using Dalamud.Plugin.Services;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using ScoutHelper.Config;
+using ScoutHelper.Models;
+using ScoutHelper.Utils;
+using ScoutHelper.Utils.Functional;
+using static ScoutHelper.Utils.Utils;
+
+namespace ScoutHelper.Managers;
+
+public class SirenManager {
+	private readonly IPluginLog _log;
+	private readonly Configuration _conf;
+
+	private readonly IDictionary<Patch, SirenPatchData> _patchData;
+	private readonly IDictionary<uint, Patch> _mobToPatch;
+
+	public SirenManager(
+		IPluginLog log,
+		Configuration conf,
+		ScoutHelperOptions options,
+		TerritoryManager territoryManager,
+		MobManager mobManager
+	) {
+		_log = log;
+		_conf = conf;
+
+		(_patchData, _mobToPatch) = LoadData(options.SirenDataFile, territoryManager, mobManager);
+	}
+
+	public (string Url, Patch HighestPatch) GenerateSirenLink(IList<TrainMob> mobList) {
+		_log.Debug("Generating a siren link for mob list: {0}", mobList);
+
+		var patches = mobList
+			.Select(mob => mob.MobId)
+			.SelectMaybe(mobId => _mobToPatch.MaybeGet(mobId))
+			.Distinct()
+			.Order()
+			.ToImmutableList();
+
+		_log.Debug("Patches represented in mob list: {0}", patches);
+
+		var fullPath = patches
+			.Select(
+				patch => {
+					var patchData = _patchData[patch];
```

Comment:
I can't test ATM but if someone puts a nonexistent patch name in here will the plogon blow up or gracefully return an error message?

Reply:
Nvm this is not an importer. My brain is potato.

Similar error could happen with DT, but I'm guessing it won't since the mob data needs to exist first